### PR TITLE
Switch to Ubuntu for docker

### PR DESCRIPTION
**Pull request by Sir-Boops**
* Based on my image [Here](https://git.sergal.org/Sir-Boops/docker-mastodon/src/branch/master/Dockerfile)
* 200~ish MB smaller then the current build ( When built right )
* Build with jemalloc for slightly less ram usage
* /Should/ be an exact drop in replacement ( I've been told by other that switched to using this image that's a complete drop in replacement and I don't see any reason it wouldn't be but another confirmation would be nice :smile:  )
* Only real downfall of this image is it has to be built using --squash ( docker beta feature ) or the image size will be huge from all the individual RUNs

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,89 +1,85 @@
...
-RUN addgroup -g ${GID} mastodon && adduser -h /mastodon -s /bin/sh -D -G mastodon -u ${UID} mastodon \
- && mkdir -p /mastodon/public/system /mastodon/public/assets /mastodon/public/packs \
- && chown -R mastodon:mastodon /mastodon/public
-
-COPY . /mastodon
-
-RUN chown -R mastodon:mastodon /mastodon
-
-VOLUME /mastodon/public/system
+FROM sirboops/nodejs:8.15.0 as node
```
Comment (Gargron) [inline Dockerfile]: Why sirboops namespace instead of official images?

Comment (Sir-Boops) [reply Dockerfile]: Just so everything is built using the same libs

Official images afaik use alpine/Deb not Ubuntu

Comment (Sir-Boops) [reply Dockerfile]: If you want everything built here I can switch it over to (this)[https://git.sergal.org/Sir-Boops/docker-mastodon/src/commit/f5faec868dfdd3c23c17d9bb1fbb6c701325db77/Dockerfile]

Just means alot longer build times

Comment (Gargron) [reply Dockerfile]: In for a penny, in for a pound. Let's switch to that. Just to have more confidence over long-term availability of these images.

**Comment by Sir-Boops** [comment]
Updated to build everything locally

I haven't tested this exact build on a live server yet but everything built and assets compiled just fine locally :+1:

File: Dockerfile [hunk 2]
```diff
@@ -1,89 +1,124 @@
...
+# Add more PATHs to the PATH
+ENV PATH="${PATH}:/opt/ruby/bin:/opt/node/bin:/opt/mastodon/bin"
+
+# Create the mastodon user
+RUN apt update && \
+	echo "Etc/UTC" > /etc/localtime && \
+	ln -s /opt/jemalloc/lib/* /usr/lib/ && \
+	apt -y dist-upgrade && \
+	apt install -y whois && \
+	addgroup --gid 991 mastodon && \
```
Comment (Gargron) [inline Dockerfile]: Missing UID and GID as args, hardcoded 991

**Review by Gargron — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (Sir-Boops) [reply Dockerfile]: Added and set the default to 991

**Review by Gargron — APPROVED** [review]

**Comment by ashleyhull-versent** [comment]
Thanks for the work @Sir-Boops. I agree with the local building method over prebaked images :)

**Pull request merged**